utils/utils.jsx

```diff
@@ -1358,55 +1355,49 @@ export function getPasswordConfig(license, config) {
 }
 
 export function isValidPassword(password, passwordConfig) {
-    let errorMsg = '';
     let errorId = 'user.settings.security.passwordError';
-    let error = false;
-    let minimumLength = Constants.MIN_PASSWORD_LENGTH;
+    let valid = true;
+    const minimumLength = passwordConfig.minimumLength || Constants.MIN_PASSWORD_LENGTH;
 
-    if (passwordConfig.isEnterprise && passwordConfig.isLicensed && passwordConfig.isPasswordRequirements) {
-        if (password.length < passwordConfig.minimumLength || password.length > Constants.MAX_PASSWORD_LENGTH) {
-            error = true;
-        }
-
-        if (passwordConfig.requireLowercase) {
-            if (!password.match(/[a-z]/)) {
-                error = true;
-            }
+    if (password.length < minimumLength || password.length > Constants.MAX_PASSWORD_LENGTH) {
+        valid = false;
+    }
 
-            errorId += 'Lowercase';
+    if (passwordConfig.requireLowercase) {
+        if (!password.match(/[a-z]/)) {
+            valid = false;
         }
 
-        if (passwordConfig.requireUppercase) {
-            if (!password.match(/[A-Z]/)) {
-                error = true;
-            }
+        errorId += 'Lowercase';
+    }
 
-            errorId += 'Uppercase';
+    if (passwordConfig.requireUppercase) {
+        if (!password.match(/[A-Z]/)) {
+            valid = false;
         }
 
-        if (passwordConfig.requireNumber) {
-            if (!password.match(/[0-9]/)) {
-                error = true;
-            }
+        errorId += 'Uppercase';
+    }
 
-            errorId += 'Number';
+    if (passwordConfig.requireNumber) {
+        if (!password.match(/[0-9]/)) {
+            valid = false;
         }
 
-        if (passwordConfig.requireSymbol) {
-            if (!password.match(/[ !"\\#$%&'()*+,-./:;<=>?@[\]^_`|~]/)) {
-                error = true;
-            }
+        errorId += 'Number';
+    }
 
-            errorId += 'Symbol';
+    if (passwordConfig.requireSymbol) {
+        if (!password.match(/[ !"\\#$%&'()*+,-./:;<=>?@[\]^_`|~]/)) {
+            valid = false;
         }
 
-        minimumLength = passwordConfig.minimumLength;
-    } else if (password.length < Constants.MIN_PASSWORD_LENGTH || password.length > Constants.MAX_PASSWORD_LENGTH) {
-        error = true;
+        errorId += 'Symbol';
```

Comment:
What happens if the password fails on multiple counts? Looks like the `errorId` isn't going to match a real string.

Reply:
We have all 15 combinations here - https://github.com/mattermost/mattermost-webapp/blob/master/i18n/en.json#L3107-L3121